### PR TITLE
test: use audit tests to compare composition outputs

### PR DESCRIPTION
**Pull request by obmarg**
Runnable with

```sh
CLICOLOR_FORCE=1 cargo nextest run -p federation-audit-tests  --no-fail-fast --test composition_comparisons --run-ignored all
```

### CODE REVIEW

File: engine/crates/engine-v2/codegen/src/loader.rs [hunk 1]
```diff
@@ -226,7 +226,7 @@ fn parse_union_kind(name: &str, directives: Iter<'_, Directive<'_>>) -> domain::
         }
     } else {
         domain::UnionKind::Record(domain::RecordUnion {
-            indexed: parse_indexed(name, directives),
+            indexed: parse_indexed(name, directives.clone()),
```
Comment (yoav-lavi) [inline engine/crates/engine-v2/codegen/src/loader.rs:229]: Do we need this clone?

**Review by tomhoule — APPROVED** [review]
:fire:

**Review by yoav-lavi — APPROVED** [review]
Looks good, see comment

File: engine/crates/engine-v2/codegen/src/loader.rs (hunk 1 above)
Comment (yoav-lavi) [reply engine/crates/engine-v2/codegen/src/loader.rs:229]: As in do we want to add a clone here specifically for testing? It's probably a non issue but thought I'd ask

Comment (obmarg) [reply engine/crates/engine-v2/codegen/src/loader.rs:229]: We're not adding a clone specifically for testing.  We are adding a clone because I updated `cynic-parser` and `Iter` (which directives is an instance of) is no longer `Copy`.  This code will not compile if I remove the `clone`.  So yes we need this clone.

It is incredibly inexpensive, and this is also just codegen code that is not performance critical.  

tldr: yes

Comment (yoav-lavi) [reply engine/crates/engine-v2/codegen/src/loader.rs:229]: I see, was wondering why it changed. Ok sounds good
